src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -734,11 +733,32 @@ def all(self) -> None:
         -----
         Not a Number (NaN), positive infinity and negative infinity
         evaluate to True because these are not equal to zero.
+
+        Examples
+        --------
+        True, because nonzero integers are considered True.
+
+        >>> pd.Index([1, 2, 3]).all()
+        True
+
+        False, because 0 is considered False.
+
+        >>> pd.Index([0, 1, 2]).all()
+        False
         """
-        # TODO: SNOW-1458141 implement all
+        # If self is empty, the result for all is always True. This `self.empty` check results in one extra query.
+        # Passing an empty object through `query_compiler.all` results in the ValueError:
+        # The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
+        if self.empty:
+            return True
+        return (
+            self.to_series()
+            ._query_compiler.all(axis=0, bool_only=None, skipna=None, **kwargs)
```

Comment:
why not just to_series().all(...)?

Reply:
That should work, I didn't think of that.